io_mesh_3mf/export_3mf.py

```diff
@@ -182,8 +222,8 @@ def unit_scale(self, context):
         """
         scale = self.global_scale
 
-        if context.scene.unit_settings.scale_length != 0:
-            scale *= context.scene.unit_settings.scale_length  # Apply the global scale of the units in Blender.
+        #if context.scene.unit_settings.scale_length != 0:
+        #    scale *= context.scene.unit_settings.scale_length  # Apply the global scale of the units in Blender.
```

Comment:
This means that the unit scale in Blender has no effect any more on how big the model becomes that gets imported. That is a bug.

3MF specifies that the model coordinates are always in millimetres. So the actual size of the model is known and should be loaded in correctly. STL has no such specification. The units are units. Pretty much all STL files assume that the unit is millimetres, although some STL files are generated in inches as well. For STL then it is important that the unit and/or scale is set correctly when importing the STL. Sadly, Blender defaults to using metres as units, which is almost always wrong. That is an issue with the STL importer though and should be fixed there.

Making STL and 3MF import at the same scale by removing the scaling from 3MF effectively makes both the STL and the 3MF files import wrongly (STL semi-wrong since it doesn't specify the unit length, and 3MF definitely wrong).